### DISCUSSION TITLE
How to add i18n?

### DISCUSSION THREAD

**Discussion by aismai** [bukhalo/nestjs-telegraf#813] [Q&A]
nestjs-i18n - throws an error

can't resolve i18n context for type: telegraf not supported yet)

**Comment by evilsprut** [comment]
I see no problem using either of the ready-made packages nestjs-i18n, or telegraf-i18n, both of which will work fine. Maybe you can give some context? Can you show the code and what your problem is?

**Comment by sgerodes** [comment]
@aismai You can disable the warning message by setting:
```
I18nModule.forRoot({
      ...
      logging: false
      ...
})
```
This does not solve the internationalization problem but disables the warning log.

**Comment by sgerodes** [comment]
In addition to the solutions mentioned by @evilsprut, you can use [nestjs-telegraf-i18n](https://www.npmjs.com/package/nestjs-telegraf-i18n), it will combine the contexts of nestjs-telegraf and nestjs-i18n.

It makes the i18n context fully aware of the Telegraf context, allowing you to access translations directly inside your bot handlers.
E.g.
```
@Start()
async start_command(@Ctx() ctx: TelegrafI18nContext) {
    const internationalized_message = ctx.t("i18n.menus.hello.message");
    await ctx.reply(internationalized_message);
}
```

More examples are in the description of the package: https://www.npmjs.com/package/nestjs-telegraf-i18n
